src/test/java/teammates/test/cases/pagedata/StudentFeedbackResultsPageDataTest.java

```diff
@@ -96,19 +96,33 @@ public void testAll() throws EntityDoesNotExistException {
         assertEquals("", questionBundle1.getQuestionDetails().getAdditionalInfo());
         assertEquals("", questionBundle2.getQuestionDetails().getAdditionalInfo());
 
-        assertNotNull(questionBundle1.getResponseTables());
-        assertNotNull(questionBundle2.getResponseTables());
+        ______TS("giver & receiver in self/others response tables");
 
-        assertEquals("You", questionBundle1.getResponseTables().get(0).getRecipientName());
+        assertNotNull(questionBundle1.getSelfResponseTables());
+        assertNotNull(questionBundle1.getOthersResponseTables());
+        assertNotNull(questionBundle2.getSelfResponseTables());
+        assertNotNull(questionBundle2.getOthersResponseTables());
 
-        assertNotNull(questionBundle1.getResponseTables().get(0).getResponses());
-
-        assertEquals("You", questionBundle1.getResponseTables().get(0).getResponses()
+        assertTrue(questionBundle1.getSelfResponseTables().get(0).isGiverNameYou());
+        assertEquals("You", questionBundle1.getSelfResponseTables().get(0).getRecipientName());
+        assertNotNull(questionBundle1.getSelfResponseTables().get(0).getResponses());
+        assertEquals("You", questionBundle1.getSelfResponseTables().get(0).getResponses()
                                         .get(0).getGiverName());
-
-        assertEquals("Student 1 self feedback.", questionBundle1.getResponseTables().get(0).getResponses()
+        assertEquals("Student 1 self feedback.", questionBundle1.getSelfResponseTables().get(0).getResponses()
                                         .get(0).getAnswer());
 
+        assertEquals(0, questionBundle1.getOthersResponseTables().size());
+
+        assertEquals(1, questionBundle2.getSelfResponseTables().size());
+
+        assertFalse(questionBundle2.getOthersResponseTables().get(0).isGiverNameYou());
+        assertEquals("You", questionBundle2.getOthersResponseTables().get(0).getRecipientName());
+        assertNotNull(questionBundle2.getOthersResponseTables().get(0).getResponses());
+        assertEquals("student2 In Course1", questionBundle2.getOthersResponseTables().get(0).getResponses()
+                .get(0).getGiverName());
+        assertEquals("Response from student 2 to student 1.", questionBundle2.getOthersResponseTables().get(0)
+                .getResponses().get(0).getAnswer());
+
```

Comment:
Are there particular structures that you're following when you wrote the tests here? They feel somewhat haphazardly put together (to me)...I can get the rough sense that you're checking if the tables are correct and if the receiver and giver are displayed correctly but I find the way that the tests are written here a tad bit messy.

Reply:
There are 2 sets of `questionBundle`s here, with different example responses added inside. The tests I added here is to check the correctness of recipient names and giver names in `selfResponseTables` as well as `othersResponseTables` for each of the `questionBundle`. The tests are actually similar to the original one, it's just that there are only tests for one table of the first `questionBundle` in the original test cases, while I added tests for two tables of both `questionBundle`.